Framework/PythonInterface/mantid/plots/modest_image/modest_image.py

```diff
@@ -85,6 +96,22 @@ def get_array(self):
         """Override to return the full-resolution array"""
         return self._full_res
 
+    def _resample_array(self, xbins=None, ybins=None):
```

Comment:
I think you're right that this is the way to go with the size workspaces as we can't afford the computation of the whole image array. My concern here is that we now have the slice viewer and colorfill plots using slightly different resampling schemes. I think the slice viewer doesn't do the interpolation?

In the slice viewer there is this separate `SamplingImage` type. Would it make sense to move this from the sliceviewer to here and then use `ModestImage` for the MD workspaces and `SamplingImage` for the matrix workspaces?

Reply:
Just to clarify here. I think the interpolation scheme here is more correct in terms of what you would see once you try and do computations on the image data so I'd like the sliceviewer to be able to take advantage of that but at the same time keeping the `ModestImage` and `SamplingImage` separate (for now) so it's clear what they are doing differently.

Reply:
Just to confirm, do you think it is best if sliceviewer calls this method (or similar) to do its `resampling` or should the regular colorfill plots used the `SamplingImage`

Reply:
I was thinking that we create a `resampling_image subpackage` of `mantid.plots` parallel to `modest_image` and move the sliceviewer `SamplingImage` to that along with the `imshow` command it comes with.

The mantid.plots.axesfunctions currently just uses `modest_image` for everything 2D but maybe we use `modest_image` for MD & `sampling_image` for `MatrixWorkspace` (or maybe those those over a certain size?). The sliceviewer can then stop calling a separate imshow function and use the standard one too.